src/systems/callbacks.jl

```diff
@@ -927,8 +915,41 @@ function generate_vector_rootfinding_callback(
             end
         end
     end
-    initialize = handle_optional_setup_fn(
-        map(fn -> fn.initialize, affect_functions), SciMLBase.INITIALIZE_DEFAULT)
+    initialize = nothing
+    if has_index_cache(sys) && (ic = get_index_cache(sys)) !== nothing
+        initialize = handle_optional_setup_fn(
+            map(
+                (cb, fn) -> begin
```

Comment:
```suggestion
                function (cb, fn) 
```
anonymous functions can be written this way as well, which maybe looks slightly